src/test/kotlin/no/nav/faktureringskomponenten/controller/FakturaserieControllerIT.kt

```diff
@@ -88,23 +92,146 @@ class FakturaserieControllerIT(
         }
 
-        nyFakturaserie.startdato.shouldBe(startDatoNy)
-        nyFakturaserie.sluttdato shouldBe sluttDatoNy
         nyFakturaserie.shouldNotBeNull().status shouldBe FakturaserieStatus.OPPRETTET
         nyFakturaserie.faktura.forEach {
             it.status shouldBe FakturaStatus.OPPRETTET
         }
     }
 
+    @Test
+    fun `erstatt fakturaserie, første faktura er Bestilt, erstatter opprinnelig og lager ny`() {
+        val startDatoOpprinnelig = LocalDate.now().minusMonths(3)
+        val sluttDatoOpprinnelig = LocalDate.now().plusMonths(9)
+
+        val opprinneligFakturaserieDto = lagFakturaserieDto(
+            fakturaseriePeriode = listOf(
+                FakturaseriePeriodeDto(
+                    BigDecimal(12000),
+                    startDatoOpprinnelig,
+                    sluttDatoOpprinnelig,
+                    "Inntekt fra utlandet"
+                ),
+            )
+        )
+
+        val opprinneligFakturaserieReferanse =
+            postLagNyFakturaserieRequest(opprinneligFakturaserieDto).expectStatus().isOk.expectBody(
+                NyFakturaserieResponseDto::class.java
+            ).returnResult().responseBody!!.fakturaserieReferanse
+
+        fakturaBestillCronjob.bestillFaktura()
+
+        val nyFakturaserieDto = lagFakturaserieDto(
+            referanseId = opprinneligFakturaserieReferanse, fakturaseriePeriode = listOf(
+                FakturaseriePeriodeDto(
+                    BigDecimal(20000),
+                    startDatoOpprinnelig,
+                    sluttDatoOpprinnelig,
+                    "Inntekt fra utlandet"
+                ),
+            )
+        )
+
+        val nyFakturaserieReferanse = postLagNyFakturaserieRequest(nyFakturaserieDto).expectStatus().isOk.expectBody(
+            NyFakturaserieResponseDto::class.java
+        ).returnResult().responseBody!!.fakturaserieReferanse
+
+        fakturaBestillCronjob.bestillFaktura()
+
+        val nyFakturaserie = fakturaserieRepositoryForTesting.findByReferanseEagerly(nyFakturaserieReferanse).shouldNotBeNull()
+        val oppdatertOpprinneligFakturaserie =
+            fakturaserieRepositoryForTesting.findByReferanseEagerly(opprinneligFakturaserieReferanse)
+
+
+        oppdatertOpprinneligFakturaserie.shouldNotBeNull().erstattetMed!!.id shouldBe nyFakturaserie.id
+        oppdatertOpprinneligFakturaserie.status shouldBe FakturaserieStatus.ERSTATTET
+        oppdatertOpprinneligFakturaserie.faktura
+            .map { it.status }
+            .shouldContainExactlyInAnyOrder(
+                FakturaStatus.BESTILT,
+                FakturaStatus.KANSELLERT,
+                FakturaStatus.KANSELLERT,
+                FakturaStatus.KANSELLERT
+            )
+
+        nyFakturaserie.shouldNotBeNull().status shouldBe FakturaserieStatus.UNDER_BESTILLING
+        nyFakturaserie.faktura
+            .map { it.status }
+            .shouldContainExactlyInAnyOrder(
+                FakturaStatus.BESTILT,
```

Comment:
Bør man ikke teste avregningsfaktura et sted?
Bugen handler om at det ikke må være avregning + vanlig faktura samtidig

Reply:
Den som er bestilt her på 171 er vel avregningsfaktura?

Reply:
Det håper jeg, men det er ikke noe testen sjekker

Reply:
Er det et egen status? har snakket en del om avregning med Isa og i den testen som går fra 2016-2017 kan man vite sikkert hva det blir. Men i alle hvor man bruker relativ date vil de potensielt feile hvis man sjekker for beløp. Det brukes localdate.now i koden så man kan plutselig være i et annet år eller få endret antall på fakturalinjer

Reply:
Tenkte på `erAvregningsfaktura` på faktura (ser ikke en bedre måte å teste dette)